Sources/TraceLog/Writer.swift

```diff
@@ -0,0 +1,41 @@
+///
+///  Writer.swift
+///
+///  Copyright 2016 Tony Stone
+///
+///  Licensed under the Apache License, Version 2.0 (the "License");
+///  you may not use this file except in compliance with the License.
+///  You may obtain a copy of the License at
+///
+///  http://www.apache.org/licenses/LICENSE-2.0
+///
+///  Unless required by applicable law or agreed to in writing, software
+///  distributed under the License is distributed on an "AS IS" BASIS,
+///  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///  See the License for the specific language governing permissions and
+///  limitations under the License.
+///
+///  Created by Tony Stone on 4/23/16.
+///
+import Swift
+
+///
+/// Public protocol for writing log messages to some device or endpoint
+///
+public protocol Writer {
+    
+    /**
+     * Called when the logger needs to log an event to this logger.
+     *
+     * @param timestamp             Timestamp of the log event (number of seconds from 1970).
```

Comment:
This is defined as time from epoch, but `ConsoleWriter.swift` and `Logger.swift` use this as time from ReferenceDate. When you have to pick one, I would prefer time from epoch (1970) instead of time from referencedate (2001).